File: mono/src/symbol/function.rs
use serde::{Serialize, Deserialize};

use crate::symbol::{Declaration, Argument};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Function {
    #[serde(flatten)]
    pub declaration: Declaration,

    /// Ex kinds: (forward, stock, etc)
    pub kind: String,

    /// Return type of the function
    pub return_type: String,

    /// Arguments of the function
    pub arguments: Vec<Argument>,
}

File: schema/src/interfaces/symbol/function.ts
import { IDeclaration  } from './base';
import { IArgument } from './argument';

export interface IFunction extends IDeclaration {
    /**
     * @brief Function kind (forward, stock, etc)
     * @readonly
     */
    readonly kind: FunctionKind;

    /**
     * @brief Return type of the function
     * @readonly
     */
    readonly returnType: string;

    /**
     * @brief Arguments of the function
     * @readonly
     */
    readonly arguments: IArgument[];
}

export enum FunctionKind {
    Forward = 'forward',
    Native = 'native',
    Stock = 'stock',
    Static = 'static',
    StaticStock = 'static stock',
    Function = 'function',
}
